src/marqo/tensor_search/tensor_search.py

```diff
@@ -81,6 +81,47 @@
 logger = get_logger(__name__)
 
 
+def _get_dimension_from_model_properties(model_properties: dict):
+    """
+    Args:
+        model_properties: dict containing model properties
+    """
+    try:
+        return model_properties["dimensions"]
+    except KeyError:
+        raise errors.InvalidArgError(
+            "The given model properties must contain a 'dimensions' key"
+        )
+
+
+def _add_knn_field(ix_settings: dict):
```

Comment:
note for future work: this can be moved, alongside create_index, into a dedicated file (like `create_index.py`)